framework/include/utils/InputParameterWarehouse.h

```diff
@@ -23,7 +23,32 @@
 class InputParameters;
 class MooseApp;
 
-typedef std::map<std::string, MooseSharedPointer<InputParameters> >::iterator InputParameterIterator;
+/**
+ * A storage container for the name of parameters to be controlled
+ */
+struct InputParametersContainer
+{
+  /// The "system" name (e.g., Kernels)
+  std::string system;
+
+  /// The user-defined tag assigned to the object
+  std::string tag;
+
+  /// The "object" name (generally a sub block in an input file)
+  std::string object;
+
+  /// The name of the parameter to be controlled
+  std::string param;
+
+  /// Input file syntax provided by creating Action
+  std::string syntax;
+
+  /// Pointer to the parameters object
+  MooseSharedPointer<InputParameters> parameters;
+};
+
+
+typedef std::vector<InputParametersContainer>::iterator InputParameterIterator;
```

Comment:
Note: I didn't tell him to remove it... I was just suggesting that he move it inside the class.

Just because I'm not personally for this kind of stuff... doesn't mean it's forbidden ;-)